recipes/GigaSpeech/gigaspeech_prepare.py

```diff
@@ -0,0 +1,703 @@
+        return None
+
+    # check reading the audio file ; HF may have some corrupted files
+    try:
+        _ = torchaudio.info(audio_path)
+    except Exception as e:
+        logger.error(f"Failed reading {audio_path}: {e}")
+        return None
+
+    text = preprocess_text(row["text"], punctuation, stopwords)
+
+    if text:
+        utt_id = row["segment_id"]
+        audio_id = row["audio_id"]
+        audio_path = row["audio"]["path"]
+        speaker = row["speaker"]
+        begin_time = float(row["begin_time"])
+        end_time = float(row["end_time"])
+        duration = end_time - begin_time
+
+        row = GigaSpeechRow(
+            utt_id=utt_id,
+            audio_id=audio_id,
+            audio_path=audio_path,
+            speaker=speaker,
+            begin_time=begin_time,
+            end_time=end_time,
+            duration=duration,
+            text=text,
+        )
+
+        return row
+    else:
+        return None
+
+
+def convert_opus2wav(audio_opus_path):
+    """Convert an opus file to a wav file.
+
+    Parameters
+    ----------
+    audio_opus_path : str
+        The path to the opus file to be converted.
+
+    Returns
+    -------
+    str
+        The path to the converted wav file.
+
+    Raises
+    ------
+    subprocess.CalledProcessError
+        If the conversion process fails.
+    """
+    audio_wav_path = audio_opus_path.replace(".opus", ".wav")
+    os.system(
+        f"ffmpeg -y -i {audio_opus_path} -ac 1 -ar {SAMPLING_RATE} {audio_wav_path} > /dev/null 2>&1"
+    )
+    return audio_wav_path
+
+
+def preprocess_text(text: str, punctuation: bool, stopwords) -> str:
+    """
+    Preprocesses the input text by removing garbage tags and removing punctuation
+    and filler words if specified.
+
+    Parameters
+    ----------
+    text : str
+        The input text to be preprocessed.
+    punctuation : bool
+        Keeping punctuation or not. Default is no.
+    stopwords : list
+        List of words to remove from the input test string.
+
+    Returns
+    -------
+    str
+        The preprocessed text with removed garbage tags and replaced punctuation tags.
+
+    Raises
+    ------
+    AssertionError
+        If '<' or '>' tags are found in the text after preprocessing.
+
+    Notes
+    -----
+    The function iterates over predefined garbage utterance tags (GARBAGE_UTTERANCE_TAGS)
+    and removes them from the input text. It then iterates over predefined punctuation tags
+    (PUNCTUATION_TAGS) and replaces them with the corresponding punctuation.
+
+    Examples
+    --------
+    >>> text = " DOUGLAS MCGRAY IS GOING TO BE OUR GUIDE YOU WALK THROUGH THE DOOR <COMMA> YOU SEE THE RED CARPETING <COMMA> YOU SEE SOMEONE IN A SUIT <PERIOD> THEY MAY BE GREETING YOU <PERIOD>"
+    >>> preprocess_text(text)
+    "douglas mcgray is going to be our guide you walk through the door, you see the red carpeting, you see someone in a suit. they may be greeting you."
```

Comment:
Changing the example to demonstrate `punctuation=True`, since otherwise this would not actually change anything in the input. (Maybe should just pick a better example)

```suggestion
    >>> preprocess_text(text, punctuation=True, stopwords=GARBAGE_UTTERANCE_TAGS)
    "DOUGLAS MCGRAY IS GOING TO BE OUR GUIDE YOU WALK THROUGH THE DOOR, YOU SEE THE RED CARPETING, YOU SEE SOMEONE IN A SUIT. THEY MAY BE GREETING YOU."
```